mlflow/projects/utils.py

```diff
@@ -96,18 +93,12 @@ def _is_git_repo(path):
 
 def _is_local_uri(uri):
     """Returns True if passed-in URI should be interpreted as a path on the local filesystem."""
-    if _GIT_URI_REGEX.match(uri):
-        return False
-
-    parsed_uri = urllib.parse.urlparse(uri)
-    drive = pathlib.Path(uri).drive
-
-    if drive != "" and drive.lower()[0] == parsed_uri.scheme:
-        return not _is_git_repo(uri)
-    elif parsed_uri.scheme in ("file", ""):
-        return not _is_git_repo(parsed_uri.path)
+    if _is_file_uri(str(uri)):
+        parsed_uri = urllib.parse.urlparse(str(uri))
+        resolved_uri = pathlib.Path(parsed_uri.netloc, parsed_uri.path).resolve()
     else:
-        return False
+        resolved_uri = pathlib.Path(uri).resolve()
+    return resolved_uri.exists()
```

Comment:
This should now take care of file URI types as well (and would be platform independent)

```bash
>>> from mlflow.projects.utils import _is_local_uri
>>> _is_local_uri(Path.cwd().as_uri())
True
>>> _is_local_uri(Path.cwd())
True
>>> _is_local_uri(Path(Path.cwd(), "setup.py").as_uri())
True
>>> _is_local_uri("https://github.com/mlflow/mlflow")
False
>>> _is_local_uri("git@github.com:mlflow/mlflow.git")
False
```